src/utils.ts

```diff
@@ -35,8 +44,126 @@ export class Utils {
     private static readonly CLI_VERSION_ARG: string = 'version';
     // Download repository input
     private static readonly CLI_REMOTE_ARG: string = 'download-repository';
+    // OpenID Connect audience input
+    private static readonly OIDC_AUDIENCE_ARG: string = 'aud';
+    //OpenID Connect provider_name input
+    private static readonly OIDC_INTEGRATION_PROVIDER_NAME : string = 'provider_name'
+
+    /**
+     * Verifies that required credentials can be collected and return the credential collection mode:
+     * CREDENTIALS_ENV_MODE: if credentials are found in JF_ENV_* environment variables
+     * CREDENTIALS_URL_MODE: if JF_URL exists and the rest of the credentials are found in JF_ACCESS_TOKEN / JF_USER + JF_PASSWORD or a new access token will be generated using OpenID-Connect flow
+     */
+    public static assertCliEnvAndReturnCredCollectionMode(): string {
+        for (let configToken of Utils.getConfigTokens()) {
+            let serverObj = JSON.parse(Buffer.from(configToken, 'base64').toString());
+            if (serverObj && serverObj.artifactoryUrl) {
+                // At this mode all credentials are to be found in on of JF_ENV_* environment variable's value
+                return Utils.CREDENTIALS_ENV_MODE;
+            }
+        }
+        if (process.env.JF_URL) {
+            // At this mode all credentials are to be collected from environment variables: JF_USER, JF_PASSWORD, JF_ACCESS_TOKEN
+            return Utils.CREDENTIALS_URL_MODE;
+        }
+        throw new Error("no JF_ENV_* env variable contains a valid url, nor JF_URL. Please provide a valid url to the server in either way")
+    }
+
+    /**
+     * Retrieves server credentials for accessing JFrog's server
+     * searching for existing environment variables such as JF_ACCESS_TOKEN or the combination of JF_USER and JF_PASSWORD.
+     * If neither is found, and if the request and requester are authorized, it generates an access token for the specified JFrog's server using the OpenID Connect mechanism.
+     */
+    public static async getJfrogCredentials(): Promise<JfrogCredentials> {
+        let jfrogCredentials : JfrogCredentials = this.collectJfrogCredentialsFromEnvVars();
+        if (!jfrogCredentials.jfrogUrl) {
+            return jfrogCredentials
+        }
+
+        // If the required credentials, such as the access token or a combination of username and password, are available, the process terminates without triggering the OIDC flow
+        if (jfrogCredentials.accessToken || (jfrogCredentials.username && jfrogCredentials.password)) {
+            return jfrogCredentials;
+        }
+
+        console.log("JF_ACCESS_TOKEN and JF_USER + JF_PASSWORD weren't found. Getting access token using OpenID Connect")
```

Comment:
I saw plenty of `console.log`. Let's replace them by core.info and core.debug.
```suggestion
        core.debug("JF_ACCESS_TOKEN and JF_USER + JF_PASSWORD weren't found. Getting access token using OpenID Connect")
```

Reply:
Sure! but can you explain why? I though its best to give as much info to the user about the OIDC process